tests/integration/test_multi_relations.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+DB_METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+SCALE_OUT = 7
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    """Build the charm and deploy 1 units to ensure a cluster is formed."""
+    # Build and deploy charm from local source folder
+    db_charm = await ops_test.build_charm(".")
+
+    config = {"profile": "testing"}
+    resources = {"mysql-image": DB_METADATA["resources"]["mysql-image"]["upstream-source"]}
+
+    await ops_test.model.deploy(
+        db_charm,
+        application_name="mysql",
+        config=config,
+        num_units=3,
+        resources=resources,
+        series="jammy",
+        trust=True,
+    )
+
+    for i in range(1, SCALE_OUT + 1):
+        await ops_test.model.deploy(
+            "mysql-test-app",
+            application_name=f"app{i}",
+            num_units=1,
+            channel="latest/edge",
```

Comment:
auto_start_writes=false ? re: performance of GH Runners.... 
P.S. consider to add rate-limit for mysql-test-app to do NOT kill DB in such cases: https://github.com/canonical/postgresql-test-app/blob/main/config.yaml#L11-L15

So you will be able to use here `sleep_interval=1000(ms)` to rate-limit 1 insert per second from each test-app.